test/redirects.js

```diff
@@ -48,9 +48,9 @@ const HTTP_TESTS = [
   ['http://url.spec.whatwg.org/', 301, 'https://url.spec.whatwg.org/', 'keep'],
   ['http://validator.whatwg.org/', 301, 'https://validator.whatwg.org/', 'keep'],
   ['http://webvtt.spec.whatwg.org/', 301, 'https://webvtt.spec.whatwg.org/', 'keep'],
-  ['http://whatwg.org/', 301, 'http://www.whatwg.org/', 'keep'], // TODO
+  ['http://whatwg.org/', 301, 'https://whatwg.org/', 'keep'],
   ['http://wiki.whatwg.org/', 302, 'https://wiki.whatwg.org/', 'keep'],
-  ['http://www.whatwg.org/foo', 404], // TODO
+  ['http://www.whatwg.org/', 301, 'https://www.whatwg.org/', 'keep'],
```

Comment:
Why not make this skip straight to no-www?

Reply:
That would violate HSTS preload guidelines.

Reply:
Basically, you first redirect to the identical secure URL and then do whatever else. It makes some sense as that's what HSTS does too internally. This is also why I was interested in testing `http://whatwg.org/c`.

Reply:
Also, it's a lot easier to always redirect to exactly the same resource on HTTPS.